src/browser/hooks/useAnimationEnd.js

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+function useAnimationEnd(
+  ref,
+  callback,
+  animationName,
+  allowEventBubbling = true,
+) {
+  React.useEffect(() => {
+    if (!ref.current) {
+      return;
+    }
+
+    function handleAnimationend(event) {
+      if (!allowEventBubbling && event.target !== ref.current) {
```

Comment:
TL;DR, I think the name would probably be better as `listenToEventBubbling` as it's not about allowing/preventing, but listening to events for all descendent elements vs just the targeted element and ignoring events from descendent elements. I will update the name accordingly. Thx.

More specifically:

This prop is not about how bubbling works or about preventing it but rather listening to it or not. I added this option thinking about broader use cases than this PR as hooks are a functional mechanism to be able to reuse code in multiple components. Scenarios are as follows:

If there is a CSS animation reused on multiple elements that are part of a shared ancestry (i.e. one element is the descendent of another) ...
   1. In one instance, I only want to know if the animation has completed on the targeted (ancestor) element and not a descendent element. In that case I can set the hook prop to not listen to event bubbling, thereby ignoring events from descendent elements that might complete first and only responding to events directly from the targeted element.
   2. In another instance, it might be easier to listen for the completion of an animation event on a descendent element from the targeted (ancestor) element because of multiple levels of components or because descendent elements are added/removed continually so registering listeners becomes a challenge to manage or there are numerous elements that all use the same animation where registering a single listener on the ancestor element is more efficient.

Does that make sense? Regardless, I'll change the name of the prop.

Reply:
ah! then it makes sense, but I think we should add it to the effect's documentation as I'm not sure people will get it right away. 

This makes me think it might make sense to have a common effects library like the common components? If so, this might be far more useful in the webapp

Reply:
True and agreed. I usually like to add JSDoc's but completely forgot for this PR 😳

RE: the library, yes we absolutely could consider that.

Reply:
I added some ESDocs